packages/web-forms/src/components/controls/Input/InputGeopoint.vue

```diff
@@ -0,0 +1,499 @@
+<script setup lang="ts">
+import InputGeopointReadonly from '@/components/controls/Input/InputGeopointReadonly.vue';
+import type { GeopointInputNode, InputValue } from '@getodk/xforms-engine';
+import Button from 'primevue/button';
+import PrimeDialog from 'primevue/dialog';
+import PrimeProgressSpinner from 'primevue/progressspinner';
+import { computed, inject, ref } from 'vue';
+
+interface InputGeopointProps {
+	readonly question: GeopointInputNode;
+}
+
+const props = defineProps<InputGeopointProps>();
+const submitPressed = inject<boolean>('submitPressed');
+const isInvalid = computed(() => props.question.validationState.violation?.valid === false);
+const coords = ref<GeolocationCoordinates | null>(null);
+const showDialog = ref(false);
+const disabled = computed(() => props.question.currentState.readonly === true);
+const geoLocationError = ref<boolean>(false);
+const watchID = ref<number | null>(null);
+const startTime = ref<number | null>(null);
+const elapsedTime = ref(0);
+// TODO: fix TypeScript check so it doesn't take types from NodeJS
+let intervalID: NodeJS.Timeout | undefined;
+
+const value = computed((): InputValue<'geopoint'> => {
+	return props.question.currentState.value;
+});
+
+/**
+ * Default accuracy in meters that can usually be reached by modern devices given enough time.
+ */
+const ACCURACY_THRESHOLD_DEFAULT = 5;
+const accuracyThreshold = computed<number>(() => {
+	return props.question.nodeOptions.accuracyThreshold ?? ACCURACY_THRESHOLD_DEFAULT;
+});
+
+/**
+ * Default unacceptable accuracy in meters, which is about the length of a city block.
+ */
+const UNACCEPTABLE_ACCURACY_THRESHOLD_DEFAULT = 100;
+const unacceptableAccuracyThreshold = computed<number>(() => {
+	const threshold = props.question.nodeOptions.unacceptableAccuracyThreshold;
+	return threshold ?? UNACCEPTABLE_ACCURACY_THRESHOLD_DEFAULT;
+});
+
+/**
+ * Defines geolocation accuracy levels:
+ *  - 'good' if accuracy is <= unacceptableAccuracyThreshold
+ *  - 'poor' if accuracy > unacceptableAccuracyThreshold
+ *  - 'unknown' if accuracy is null or undefined
+ */
+const ACCURACY_QUALITY = {
+	GOOD: 'GOOD',
+	POOR: 'POOR',
+	UNKNOWN: 'UNKNOWN',
+} as const;
+
+type AccuracyQualityEnum = typeof ACCURACY_QUALITY;
+
+type AccuracyQualityValue = AccuracyQualityEnum[keyof AccuracyQualityEnum];
+
+const getQualityCoordinates = (accuracy: number | null | undefined): AccuracyQualityValue => {
+	if (accuracy == null) {
+		return ACCURACY_QUALITY.UNKNOWN;
+	}
+
+	if (accuracy > unacceptableAccuracyThreshold.value) {
+		return ACCURACY_QUALITY.POOR;
+	}
+
+	return ACCURACY_QUALITY.GOOD;
+};
+
+const getQualityLabel = (quality: AccuracyQualityValue): string => {
+	// TODO: translations
```

Comment:
Since this applies to each of the return values:

```suggestion
// TODO: translations
const getQualityLabel = (quality: AccuracyQualityValue): string => {
```

Reply:
Fixed